### PR TITLE
Add coommon test of the key PERFORMANCE_HINT for AUTO plugin API 2.0.

### PR DESCRIPTION
**Pull request by yangwang201911**
Signed-off-by: Wang, Yang <yang4.wang@intel.com>
Add common test case for AUTO plugin(MULTI) only.
### Details:
 - Add OV 2.0 test case covering the key PERFORMANCE_HINT for AUTO Plugin.

### Tickets:
 - https://jira.devtools.intel.com/browse/CVS-78646

### CODE REVIEW

**Review by myshevts — APPROVED** [review]

File: src/tests/functional/plugin/gpu/shared_tests_instances/behavior/ov_plugin/core_integration.cpp [hunk 1]
```diff
@@ -88,10 +88,16 @@ INSTANTIATE_TEST_SUITE_P(
...
         {ov::device::priorities(CommonTestUtils::DEVICE_CPU)},
         {ov::device::priorities(CommonTestUtils::DEVICE_GPU)},
-        {ov::device::priorities(CommonTestUtils::DEVICE_CPU, CommonTestUtils::DEVICE_GPU)}};
+        {ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT)},
+        {ov::hint::performance_mode(ov::hint::PerformanceMode::LATENCY)},
+        {ov::hint::performance_mode(ov::hint::PerformanceMode::UNDEFINED)},
+        {ov::device::priorities(CommonTestUtils::DEVICE_GPU), ov::hint::performance_mode(ov::hint::PerformanceMode::UNDEFINED)},
+        {ov::device::priorities(CommonTestUtils::DEVICE_GPU), ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT)},
+        {ov::device::priorities(CommonTestUtils::DEVICE_GPU), ov::hint::performance_mode(ov::hint::PerformanceMode::LATENCY)},
+        {ov::device::priorities(CommonTestUtils::DEVICE_GPU, CommonTestUtils::DEVICE_GPU)}};
```
Comment (songbell) [inline src/tests/functional/plugin/gpu/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:97]: How about CPU,GPU with performance hint?

File: src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/core_integration.cpp [hunk 2]
```diff
@@ -81,10 +81,17 @@ INSTANTIATE_TEST_SUITE_P(
         ::testing::Values("MULTI", "AUTO"));
 
 const std::vector<ov::AnyMap> multiConfigs = {
-        {ov::device::priorities(CommonTestUtils::DEVICE_CPU)}};
+        {ov::device::priorities(CommonTestUtils::DEVICE_CPU)},
+        {ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT)},
```
Comment (songbell) [inline src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:85]: will this load GPU in auto?

Comment (yangwang201911) [reply src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:85]: Will update to load to CPU only in Auto.

File: src/tests/functional/plugin/gpu/shared_tests_instances/behavior/ov_plugin/core_integration.cpp (hunk 1 above)
Comment (yangwang201911) [reply src/tests/functional/plugin/gpu/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:97]: Will update.

**Review by myshevts — APPROVED** [review]

**Pull request merged**